conbench/entities/run.py

```diff
@@ -186,6 +147,11 @@ def get_default_baseline_run(
         """
         from ..entities.benchmark_result import BenchmarkResult
 
+        if self.commit is None:
+            # No associated repo/commit information. Raise an exception
+            # instead?
+            return None
```

Comment:
Another occasion where I think type checking revealed a special case that wants to be handled _here_. I don't know if this was before handled deeper into the business logic, but I think here really is a good place to handle this case, towards code readability.

But maybe there was even a bug hiding here?

Reply:
This case was handled in the try-caught `this_commit.commit_ancestry_query` property, and turned into a no-op with a debug message. Perhaps add the debug message back here?

Reply:
Thanks for adding detail.

> was handled

Did I change this in this PR?

Looking at https://github.com/conbench/conbench/pull/1134/files#diff-8e7df94d5fa6b0082305db878bc76670c59fc020792fb4f78a494d083faaedeaR158 I am not so sure.

Let's leave this as a follow-up item.

Reply:
Before this PR, `self.commit` was guaranteed to be a `Commit` instance—never None—because the foreign key was non-nullable. 

If the run truly wasn't associated with a commit, `self.commit` would have been one of those "no-context" commits with the branch/fork_point/timestamp properties empty. So the `commit_ancestry_query` property would have failed here:

https://github.com/conbench/conbench/blob/a2732d3ee415cdcf0b342f73a15e35110739d6a1/conbench/entities/commit.py#L203-L204

Reply:
Thank you for clarifying! Now I understand. 

Do we want to retain the `CantFindAncestorCommitsError` exception type for now? 

> Perhaps add the debug message back here?

Yes, and also we can maybe make the error propagation better and feed all the way to the user (as you also said somewhere in this discussion) the _reason_ for why history analysis couldn't be performed -- the relevant error detail here would be something like "no commit information stored with this benchmark result, cannot inspect commit ancestry".

Reply:
>Do we want to retain the CantFindAncestorCommitsError exception type for now?

Yes, I think that might be a good idea, at least for the case when we're trying to examine a partially-filled commit (i.e. one with a commit hash but no metadata; to be filled in later).

>feed all the way to the user (as you also said somewhere in this discussion) the reason for why history analysis couldn't be performed

Agreed!

Reply:
>feed all the way to the user (as you also said somewhere in this discussion) the reason for why history analysis couldn't be performed

I am going to do this in https://github.com/conbench/conbench/pull/1078 (at least when trying to get contender baseline run IDs).